### PR TITLE
Fixes #32893 - use setting DSL for General category

### PR DESCRIPTION
**Pull request by ezr-ondrej**
This uses our new DSL to define the General settings, this showcases how
to use the new DSL and other categories should follow in the same
release.

### CODE REVIEW

File: app/registries/foreman/setting_manager.rb [hunk 1]
```diff
@@ -44,11 +50,13 @@ def initialize(storage, context_name, category_name)
       #               default: true,
       #               description: N_('Use Puppet that goes to 11'),
       #               full_name: N_('Use shiny puppet'),
-      #               encrypt: true)
+      #               encrypt: true
```
Comment (ekohl) [inline app/registries/foreman/setting_manager.rb]: Missing comma here

Comment (ezr-ondrej) [reply app/registries/foreman/setting_manager.rb]: Indeed, thanks, fixed in the origin PR #8470

**Comment by ares** [comment]
Nice, I like the DSL

**Comment by ezr-ondrej** [comment]
[test foreman]

**Comment by ezr-ondrej** [comment]
I've splitted the fixes necessary for this to work into separate commit.

**Review by ares — APPROVED** [review]
Works fine, I didn't find any breaking change. I also scanned other repos for direct use of `Setting::General` but I didn't find anything. The down migration is not necessary, if someone wants to get to previous version, they must reset the code and then settings gets reseeding automatically.

Great work, thanks @ezr-ondrej, merging!

**Pull request merged**